src/main/java/org/opensearch/security/tools/SecurityAdmin.java

```diff
@@ -1218,6 +1220,9 @@ private static int upload(RestHighLevelClient tc, String index, String cd, boole
         if (new File(cd+"audit.yml").exists()) {
             success = uploadFile(tc, cd + "audit.yml", index, "audit", legacy, resolveEnvVars) && success;
         }
+        if (new File(cd+"allowlist.yml").exists()) {
```

Comment:
Does this mean that `allowlist.yml` is optional, but `whitelist.yml` is required?

Reply:
Good question! `allowlist.yml` is an alternative to and will be a replacement of `whitelist.yml`, so it does not necessarily exist in a cluster.

Reply:
@cwperks Good question! `allowlist.yml` is an alternative to and will be a replacement of `whitelist.yml`, so it does not necessarily exist in a cluster.

Reply:
There's an existence check around `allowlist.yml`, but no check around `whitelist.yml` a few lines above.

```
success = uploadFile(tc, cd+"whitelist.yml", index, "whitelist", legacy, resolveEnvVars) && success;
 if (new File(cd+"allowlist.yml").exists()) {
    success = uploadFile(tc, cd + "allowlist.yml", index, "allowlist", legacy, resolveEnvVars) && success;
}
```

When I run securityadmin after removing `whitelist.yml`, but have a file called `allowlist.yml` I get the following error: 

```ERR: Seems ../../../config/opensearch-security/whitelist.yml is not in legacy format: java.io.FileNotFoundException: ../../../config/opensearch-security/whitelist.yml (No such file or directory)```

I was wondering if it made sense to check for the existence of either and fail if neither exists.

Reply:
`whitelist.yml` is required, so there's no need to check the existence.